ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -2308,16 +2255,43 @@ MiProtectVirtualMemory(IN PEPROCESS Process,
             goto FailPath;
         }
 
-        /* Check if data or page file mapping protection PTE is compatible */
-        if (!Vad->ControlArea->u.Flags.Image)
+        /* Handle legacy MM case */
+        if (Vad->u.VadFlags.Spare)
         {
-            /* Not yet */
-            DPRINT1("Fixme: Not checking for valid protection\n");
+            PMEMORY_AREA MemoryArea = (PMEMORY_AREA)Vad;
+
+            if (MemoryArea->DeleteInProgress)
```

Comment:
Can we move this to `` MmDeleteVirtualMapping`` ?

Reply:
MmProtectSectionView ? Sure.